Avoid using Write-Host and other Script-Analyzer warnings

Trenly opened the pull request:
## 📖 Description
This PR
1. Fixes all of the warnings that `Invoke-ScriptAnalyzer` was giving
2. Lints the scripts
3. Adds a workflow to report any errors for future script updates

## 🔗 References and Related Issues
* https://github.com/Trenly/winget-studio/actions/runs/19518003516
* https://github.com/microsoft/winget-pkgs/commits/master/.github/workflows/scriptAnalyzer.yaml

## 🔍 How to Test
* See first link in related issues for the workflow status
* Run a test build locally using the updated scripts

## ✅ Checklist
- [ ] Closes #xxx
- [x] Tests added/passed
	- Adding the workflow is the test to ensure the files don't have any errors added. Right now there are also no warnings. I opted to *not* fail the workflow on warnings, but it would be a simple change if we want to keep these scripts exceptionally clean.
- [ ] Documentation updated

denelon commented:
It looks like this needs to be rebased. When @AmelBawa-msft is back, I'll ask him to take a look at this.

Trenly committed:
Add workflow file

Trenly committed:
Also check modules

Trenly committed:
Lint scripts

Trenly committed:
Use unused variable

Trenly committed:
Support ShouldProcess on Remove certs

Trenly committed:
Change verb to not require ShouldProcess

Trenly committed:
Support ShouldProcess on New-AppxBundleMapping

Trenly committed:
Support ShouldProcess on New-AppxBundle

Trenly committed:
Lint

Trenly commented:
> It looks like this needs to be rebased. When @AmelBawa-msft is back, I'll ask him to take a look at this.

I've rebased and updated the implementation
